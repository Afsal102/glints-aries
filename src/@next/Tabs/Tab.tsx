import React from 'react';
import { Typography } from '../Typography';
import { StyledSpan, StyledTabButton } from './TabStyle';

export type TabProps = {
  content: React.ReactNode;
  id?: string;
  selected?: boolean;
  onSelect: () => void;
};

export const Tab = ({ content, id, selected, onSelect }: TabProps) => {
  return (
    <StyledTabButton
      data-selected={selected}
      id={id}
      onClick={() => onSelect()}
    >
      <StyledSpan>
        <Typography as="span" variant="body1">
          {content}
        </Typography>
      </StyledSpan>
    </StyledTabButton>
  );
};
