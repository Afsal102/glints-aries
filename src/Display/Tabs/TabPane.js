/* @flow */

import React from 'react';

import { TabsContent } from '../../Style/Display/TabsStyle';

const TabPane = (props: Props) => {
  const {
    children,
    className,
    ...defaultProps
  } = props;

  return (
    <TabsContent
      className={className}
      {...defaultProps}
    >
      {children}
    </TabsContent>
  );
};

type Props = {
  children: React$Node,
  className: string,
};

export default TabPane;
