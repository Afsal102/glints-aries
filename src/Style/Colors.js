export const PrimaryColor = {
  glintsred: '#EC272B',
  glintsblue: '#0BAEEC',
  glintsyellow: '#FFF240',
};

export const SecondaryColor = {
  white: '#FFFFFF',
  whitesmoke: '#F5F5F5',
  black: '#000000',
  lightblack: '#777777',
  grey: '#646464',
  lightgrey: '#C6C6C6',
  buttercup: '#F5A623',
  purplepizza: '#FF00E5',
  blue: '#0869C2',
  bluegem: '#9013FE',
};

