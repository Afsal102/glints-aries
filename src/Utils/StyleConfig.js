export const Variant = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  GHOST: 'ghost',
  LINK: 'link',
};

export const Color = {
  RED: 'red',
  YELLOW: 'yellow',
  BLUE: 'blue',
};