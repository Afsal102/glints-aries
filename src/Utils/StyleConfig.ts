export enum ETabAlignment {
  HORIZONTAL = 'horizontal',
  VERTICAL = 'vertical',
}

export enum EHorizontalTabVariant {
  UNDERLINED = 'underlined',
  COLORED = 'colored',
}

export enum ETabThemeVariant {
  BLUE = 'blue',
  BLACK = 'black',
}

export enum ETooltipPosition {
  TOP = 'top',
  BOTTOM = 'bottom',
  LEFT = 'left',
  RIGHT = 'right',
}

export const ButtonVariant = {
  SOLID_WHITE: 'solid-white',
  SOLID_BLUE: 'solid-blue',
  YELLOW: 'yellow',
  GHOST: 'ghost',
  WHITE_GREY: 'white-grey',
  LINK: 'link',
};

export const PsychedelicTheme = {
  BLUE_DOMINANT: 'blue-dominant',
  BLUE_DEFAULT: 'blue-default',
  RED_DOMINANT: 'red-dominant',
  RED_DEFAULT: 'red-default',
  YELLOW_DOMINANT: 'yellow-dominant',
  YELLOW_DEFAULT: 'yellow-default',
};

export const ButtonTheme = {
  BLUE: 'blue',
  WHITE: 'white',
};

export const ScreenSize = {
  mobileS: 320,
  mobileM: 480,
  mobileL: 640,
  tablet: 768,
  desktopS: 1024,
  desktopM: 1260,
  desktopL: 1440,
};

export const Device = {
  mobileS: `(max-width: ${ScreenSize.mobileM - 1}px)`,
  mobileM: `(min-width: ${ScreenSize.mobileM}px)`,
  mobileL: `(min-width: ${ScreenSize.mobileL}px)`,
  tablet: `(min-width: ${ScreenSize.tablet}px)`,
  desktopS: `(min-width: ${ScreenSize.desktopS}px)`,
  desktopM: `(min-width: ${ScreenSize.desktopM}px)`,
  desktopL: `(min-width: ${ScreenSize.desktopL}px)`,
};
