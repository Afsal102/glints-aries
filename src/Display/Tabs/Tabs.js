/* @flow */

import React, { Component } from 'react';

import TabPane from './TabPane';

import {
  TabsContainer,
  TabsHeader,
  TabsBody,
} from '../../Style/Display/TabsStyle';

class Tabs extends Component <Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      currentIndex: 0,
    };
  }

  handleClickTab = (index) => {
    const listener = (e) => {
      this.setState({
        currentIndex: index,
      });
    };
    return listener;
  }

  render() {
    const {
      children,
      className,
    } = this.props;

    const { currentIndex } = this.state;

    return (
      <TabsContainer className={className}>
        <TabsHeader>
          <ul role="tablist">
            {children.map((data, index) => (
              <li
                id={`tab-${index}`}
                key={data.props.tab}
                className={currentIndex === index ? 'active' : undefined}
                role="tab"
                aria-selected={currentIndex === index && true}
                aria-controls={`tab-item-${index}`}
                tabIndex={currentIndex === index ? '0' : '-1'}
              >
                <button
                  type="button"
                  onClick={this.handleClickTab(index)}
                  tabIndex="-1"
                >
                  {data.props.tab}
                </button>
              </li>
            ))}
          </ul>
        </TabsHeader>
        <TabsBody tabIndex="0">
          <TabPane
            id={`tab-item-${currentIndex}`}
            className={className}
            role="tabpanel"
            aria-labelledby={`tab-${currentIndex}`}
            tabIndex="-1"
          >
            {children[currentIndex].props.children}
          </TabPane>
        </TabsBody>
      </TabsContainer>
    );
  }
}

type Props = {
  children: React$Node,
  label: string,
  className: string,
}

type State = {
  currentIndex: number,
}

export default Tabs;
