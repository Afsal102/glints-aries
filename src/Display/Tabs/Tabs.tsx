import * as React from 'react';

import classNames from 'classnames';

import TabPane, { Props as TabPaneProps } from './TabPane';

import {
  TabsContainer,
  TabsHeader,
  TabsBody,
} from '../../Style/Display/TabsStyle';

const Tabs: Tabs = ({
  activeTab,
  onTabClick,
  children,
  variant = 'underlined',
  className,
  alignment = 'horizontal',
}) => {
  const [currentIndex, setCurrentIndex] = React.useState(0);
  const activeTabOrIndex: string | number = activeTab || currentIndex;
  const childrenArray = React.Children.toArray(children);
  const currentChild = childrenArray[currentIndex];

  const handleTabClick = (index: number, tab: React.ReactText) => {
    const listener = () => {
      setCurrentIndex(index);
      if (onTabClick) {
        onTabClick(tab);
      }
    };
    return listener;
  };

  return (
    <TabsContainer
      className={classNames(`${alignment}-aries-tabs`, 'aries-tabs', className)}
    >
      <TabsHeader
        className={classNames(`${alignment}-tabs-header`, 'tabs-header')}
      >
        <ul
          className={classNames(
            `${alignment}-tabs-list`,
            'tabs-list',
            `${variant}`
          )}
          role="tablist"
        >
          {React.Children.map(
            children,
            (data: React.ReactElement<TabPaneProps>, index) => {
              const tabLabel = data.props.label || index;
              return (
                <li
                  className={classNames(
                    `tab-${tabLabel}`,
                    { active: activeTabOrIndex === tabLabel },
                    `${alignment}-tab`,
                    `${variant}`
                  )}
                  key={data.props.tab}
                  role="tab"
                  aria-selected={activeTabOrIndex === tabLabel && true}
                  aria-controls={`tab-item-${tabLabel}`}
                  tabIndex={-1}
                >
                  <button
                    type="button"
                    onClick={handleTabClick(index, tabLabel)}
                  >
                    {data.props.tab}
                  </button>
                </li>
              );
            }
          )}
        </ul>
      </TabsHeader>
      <TabsBody className="tabs-body" tabIndex={0}>
        <TabPane
          className={classNames(
            'tabs-item',
            `tab-item-${activeTabOrIndex}`,
            `tabs-item-${alignment}`,
            'tabs-item'
          )}
          role="tabpanel"
          aria-labelledby={`tab-${activeTabOrIndex}`}
          tabIndex={-1}
        >
          {React.isValidElement<TabPaneProps>(currentChild) &&
            currentChild.props.children}
        </TabPane>
      </TabsBody>
    </TabsContainer>
  );
};

type Tabs = React.FunctionComponent<Props> & {
  Pane: typeof TabPane;
};

Tabs.Pane = TabPane;

interface Props {
  children: React.ReactNode;
  variant?: string;
  alignment?: string;
  activeTab?: string;
  onTabClick?(tab: React.ReactText): void;
  className?: string;
}

export default Tabs;
