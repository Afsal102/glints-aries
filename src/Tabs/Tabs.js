/* @flow */

import React, { Component } from 'react';
import { Container } from '../Style/TabsStyle';

class Tabs extends Component <Props, State> {
	render() {
		const { children } = this.props;

		return(
			<Container>
				<ul>
					<li className="active">test</li>
					<li>qwe</li>
					<li>123</li>
					<li>vbc</li>
					<li>321</li>
				</ul>
				{children}
			</Container>
		);
	}
}

type Props = {
  children: React$Node,
  label: string,
  className: string,
}

export default Tabs;
