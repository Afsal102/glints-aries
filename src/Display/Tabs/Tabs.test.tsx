import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';

import Tabs from './Tabs';
import { EHorizontalTabVariant } from '../../Utils/StyleConfig';
import { SecondaryColor } from '../../Style/Colors';

describe('<Tabs/> render', () => {
  test('should match snapshot', () => {
    const snapshot = renderer
      .create(
        <Tabs>
          <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
          <Tabs.Pane tab="Company">Glints</Tabs.Pane>
          <Tabs.Pane tab="Location">Jakarta</Tabs.Pane>
          <Tabs.Pane tab="Salary">Rp 10,000,000</Tabs.Pane>
        </Tabs>
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('<Tabs/> should render correctly with the alignment prop', () => {
  test('horizontal, when alignment is set horizontal', () => {
    const { getByRole } = render(
      <Tabs alignment="horizontal">
        <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
        <Tabs.Pane tab="Company">Glints</Tabs.Pane>
        <Tabs.Pane tab="Location">Jakarta</Tabs.Pane>
        <Tabs.Pane tab="Salary">Rp 10,000,000</Tabs.Pane>
      </Tabs>
    );
    const tabContainer = getByRole('tablist');
    expect(tabContainer).toHaveStyle('display: flex');
  });
});

describe('<Tabs/> should render correctly with the variant prop', () => {
  test('colored, when alignment is set colored', () => {
    const { getByRole } = render(
      <Tabs alignment="horizontal" variant={EHorizontalTabVariant.COLORED}>
        <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
        <Tabs.Pane tab="Company">Glints</Tabs.Pane>
        <Tabs.Pane tab="Location">Jakarta</Tabs.Pane>
        <Tabs.Pane tab="Salary">Rp 10,000,000</Tabs.Pane>
      </Tabs>
    );
    const tabContainer = getByRole('tablist');
    expect(tabContainer).toHaveStyle(`
      border-top: 1px solid ${SecondaryColor.lightergrey}; 
      border-bottom: 1px solid ${SecondaryColor.lightergrey};
    `);
  });

  test('underlined, when alignment is set underlined', () => {
    const { getByRole } = render(
      <Tabs alignment="horizontal" variant={EHorizontalTabVariant.UNDERLINED}>
        <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
        <Tabs.Pane tab="Company">Glints</Tabs.Pane>
        <Tabs.Pane tab="Location">Jakarta</Tabs.Pane>
        <Tabs.Pane tab="Salary">Rp 10,000,000</Tabs.Pane>
      </Tabs>
    );
    const tabContainer = getByRole('tablist');
    expect(tabContainer).toHaveStyle(
      `border-bottom: 1px solid ${SecondaryColor.lightergrey}`
    );
  });
});

describe('<Tabs/> should render correctly with an active tab item set', () => {
  const inactiveTabIndex = 0;
  const activeTabIndex = 1;
  test('active tab, to be visible when set', () => {
    const { getAllByRole } = render(
      <Tabs activeTab={activeTabIndex}>
        <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
        <Tabs.Pane tab="Company">Glints</Tabs.Pane>
      </Tabs>
    );
    const activeTab = getAllByRole('tab-button');

    expect(activeTab[activeTabIndex]).toHaveStyle('font-weight: bold');
    expect(activeTab[inactiveTabIndex]).not.toHaveStyle('font-weight: bold');
  });
});

describe('<Tabs/> should render correctly when an item is clicked', () => {
  const firstTabIndex = 0;
  const clickedTabIndex = 1;
  test('clicked tab pane should show the correct body and the item should turn active', () => {
    const { getAllByRole, queryByText } = render(
      <Tabs>
        <Tabs.Pane tab="Job">Software Engineer</Tabs.Pane>
        <Tabs.Pane tab="Company">Glints</Tabs.Pane>
      </Tabs>
    );
    const tab = getAllByRole('tab-button');

    expect(tab[firstTabIndex]).toHaveStyle('font-weight: bold');
    expect(tab[clickedTabIndex]).not.toHaveStyle('font-weight: bold');

    fireEvent.click(tab[clickedTabIndex]);
    const Job = queryByText('Software Engineering');
    const Company = queryByText('Glints');

    expect(tab[firstTabIndex]).not.toHaveStyle('font-weight: bold');
    expect(tab[clickedTabIndex]).toHaveStyle('font-weight: bold');
    expect(Job).not.toBeInTheDocument();
    expect(Company).toBeVisible();
  });
});
