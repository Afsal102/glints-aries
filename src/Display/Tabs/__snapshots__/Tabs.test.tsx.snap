// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs/> render should match snapshot 1`] = `
<div
  className="TabsStyle__TabsContainer-sc-1bcl7qu-0 djllus horizontal-aries-tabs aries-tabs"
>
  <div
    className="TabsStyle__TabsHeader-sc-1bcl7qu-1 iuuVHD horizontal-tabs-header tabs-header"
  >
    <ul
      className="horizontal-tabs-list tabs-list underlined"
      role="tablist"
    >
      <li
        aria-controls="tab-item-0"
        aria-selected={true}
        className="tab-0 active horizontal-tab underlined"
        role="tab"
        tabIndex={-1}
      >
        <button
          onClick={[Function]}
          role="tab-button"
          type="button"
        >
          Job
        </button>
      </li>
      <li
        aria-controls="tab-item-1"
        aria-selected={false}
        className="tab-1 horizontal-tab underlined"
        role="tab"
        tabIndex={-1}
      >
        <button
          onClick={[Function]}
          role="tab-button"
          type="button"
        >
          Company
        </button>
      </li>
      <li
        aria-controls="tab-item-2"
        aria-selected={false}
        className="tab-2 horizontal-tab underlined"
        role="tab"
        tabIndex={-1}
      >
        <button
          onClick={[Function]}
          role="tab-button"
          type="button"
        >
          Location
        </button>
      </li>
      <li
        aria-controls="tab-item-3"
        aria-selected={false}
        className="tab-3 horizontal-tab underlined"
        role="tab"
        tabIndex={-1}
      >
        <button
          onClick={[Function]}
          role="tab-button"
          type="button"
        >
          Salary
        </button>
      </li>
    </ul>
  </div>
  <div
    className="TabsStyle__TabsBody-sc-1bcl7qu-3 gzOTzN tabs-body"
    tabIndex={0}
  >
    <div
      aria-labelledby="tab-0"
      className="TabsStyle__TabsContent-sc-1bcl7qu-2 ezqQUq tabs-item tab-item-0 tabs-item-horizontal tabs-item"
      role="tabpanel"
      tabIndex={-1}
    >
      Software Engineer
    </div>
  </div>
</div>
`;
